### PR TITLE
Model scripting

### PR DESCRIPTION
**Pull request by sethalves**
- initial work leading to in-world mesh creation
- script interface that allows extraction of mesh from a polyvox
- script interface that allows converting a mesh to an obj asset

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5b28f3a3c427659c694146d870c5fe0406eabf11 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.dmg

**Comment by sethalves** [comment]
test this please

**Comment by marko8904** [comment]
test this please

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5b28f3a3c427659c694146d870c5fe0406eabf11 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.dmg

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5b28f3a3c427659c694146d870c5fe0406eabf11 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5b28f3a3c427659c694146d870c5fe0406eabf11 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-5b28f3a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 40b87171a9c73675244e9d6e7ebc985d10ad0a7d available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-40b8717.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-40b8717.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 08ad56d25427759a84b1fb235522da2ec1290599 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-08ad56d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-08ad56d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0785172398baef65342fe0bca25247cdf27f8c25 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-0785172.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-0785172.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7afe091c52c5ec02572347e159a25225a3f19553 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-7afe091.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-7afe091.dmg

**Comment by sethalves** [comment]
to test:
```javascript
var position = Vec3.sum(MyAvatar.position, Vec3.multiplyQbyV(MyAvatar.orientation, {x: 0, y: 0.5, z: -3.5}));

var voxelID = Entities.addEntity({
    type: "PolyVox",
    dimensions: { x: 2, y: 2, z: 2 },
    voxelVolumeSiez: { x: 32, y: 32, z: 32 },
    position: position,
    name: "polyvox mesh extract test",
    lifetime: 6000,
    dynamic: false,
    collisionless: true
});


Entities.setVoxelSphere(voxelID, position, 1, 255);

Entities.voxelsToMesh(voxelID, function(mesh) {
    print("number of vertices: " + mesh.getNumVertices());
    var objData = Model.meshToOBJ([mesh]);
    Assets.uploadData(objData, function(url, hash) {
        print("done uploading to " + url);
        Assets.setMapping("/obj-writer-test.obj", hash, function() {
            print("done setting mapping");
            Entities.addEntity({
                type: "Model",
                modelURL: "atp:/obj-writer-test.obj",
                dimensions: { x: 2, y: 2, z: 2 },
                position: Vec3.sum(position, { x: 0, y: 2, z: 0 }),
                name: "polyvox mesh extract test model",
                lifetime: 6000,
                dynamic: false,
                collisionless: true
            });
        });
    });
});
```

**Comment by Nex-Pro** [comment]
Testing this.

**Comment by Nex-Pro** [comment]
If the image below corresponds to expected result then this is an almighty Seth shaped **PASS**

hifi-snap-by-piper-peppercorn-on-2017-03-15_16-18-41

**Comment by hifi-gustavo** [comment]
Build console output for commit: 693aee062829622111b553d07a0747af243cc1fc available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-693aee0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-693aee0.dmg

File: libraries/fbx/src/OBJWriter.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    out << "\n";
+
+    // write out faces
+    int nth = 0;
+    foreach (const MeshPointer& mesh, meshes) {
+        currentVertexStartOffset = meshVertexStartOffset.takeFirst();
+
+        const gpu::BufferView& partBuffer = mesh->getPartBuffer();
+        const gpu::BufferView& indexBuffer = mesh->getIndexBuffer();
+        // const gpu::BufferView& vertexBuffer = mesh->getVertexBuffer();
```
Comment (ZappoMan) [inline libraries/fbx/src/OBJWriter.cpp]: delete dead code

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0e8d62c11e5cfa1421d8f28dd6137c6730ed4d31 available at: https://gist.github.com/210d0bc2a95d4d057292dbfe01caa902<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-0e8d62c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9885-0e8d62c.dmg

**Pull request merged**